src/helpers/findAll.ts

```diff
@@ -23,3 +27,35 @@ export function findAll(
     (element) => !isHiddenFromAccessibility(element, { cache })
   );
 }
+
+// Extracted from React Test Renderer
+// src: https://github.com/facebook/react/blob/8e2bde6f2751aa6335f3cef488c05c3ea08e074a/packages/react-test-renderer/src/ReactTestRenderer.js#L402
+function findAllInternal(
+  root: ReactTestInstance,
+  predicate: (element: ReactTestInstance) => boolean,
+  options?: FindAllOptions
+): Array<ReactTestInstance> {
+  const results: ReactTestInstance[] = [];
+
+  // Match descendants first but do not add them to results yet.
+  const matchingDescendants: ReactTestInstance[] = [];
+  root.children.forEach((child) => {
+    if (typeof child === 'string') {
+      return;
+    }
+    matchingDescendants.push(...findAllInternal(child, predicate, options));
+  });
+
+  if (
+    // Deepest only mode: add current element only if no descendants match
```

Comment:
```suggestion
    // In deepest only mode: add current element only if no descendants match
```
I would add "in" otherwise we think at first the if condition concerns only the deep only mode